lib/pltr/v2/reducers/ui.js

```diff
@@ -640,6 +642,26 @@ const updateUI = (state, action) => {
       }
     }
 
+    case PIN_PLOTLINE: {
```

Comment:
Can you please use `Math.max` and `Math.min` here to prevent the count of pinned plot lines from exceeding the number of lines or going below zero.

Reply:
@Quiescent sorry im not familiar how we change the timeline 😅

Reply:
The action is: `CHANGE_CURRENT_TIMELINE`.
You can reset the number of pinned lines there, but I actually think that'd be the wrong approach.
Better: you should an object per book to record the number of pinned timelines for each book.
i.e.:
```
{
   ...
   timeline: {
     ...
     pinnedPlotlines: {
       ...
       [bookId]: X,
     }
   }
}
```

Then you'd update the selector so that it selects from the book that's currently open.